File: android/src/main/kotlin/id/bni46/microatmsdk/data/BluetoothClient.kt
package id.bni46.microatmsdk.data

import androidx.annotation.Keep
import com.google.gson.Gson

@Keep
data class BluetoothClient(val name: String, val macAddress: String, val bonded: Boolean) {
    fun toJson(): String? {
        return Gson().toJson(this)
    }
}

File: lib/data/bluetooth_client.dart
part of '../microatmsdk.dart';

class BluetoothClient {
  late String name;
  late String macAddress;
  late bool bonded;

  BluetoothClient({
    this.name = "",
    this.macAddress = "",
    this.bonded = false,
  });
  BluetoothClient.fromJson(Map<String, dynamic> json) {
    name = json['name'];
    macAddress = json['macAddress'];
    bonded = json['bonded'];
  }
  Map<String, dynamic> toJson() {
    final data = <String, dynamic>{};
    data['name'] = name;
    data['macAddress'] = macAddress;
    data['bonded'] = bonded;
    return data;
  }
}
